src/Sylius/Bundle/ApiBundle/Validator/Constraints/PromotionEligibilityValidator.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\Constraints;
+
+use DateTime;
+use Sylius\Bundle\ApiBundle\Command\Cart\ApplyCouponToCart;
+use Sylius\Component\Promotion\Model\PromotionCouponInterface;
+use Sylius\Component\Promotion\Repository\PromotionCouponRepositoryInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Webmozart\Assert\Assert;
+
+/** @experimental */
+final class PromotionEligibilityValidator extends ConstraintValidator
```

Comment:
Instead of these two validators we should use `\Sylius\Component\Promotion\Checker\Eligibility\CompositePromotionEligibilityChecker`. Sample constraint may be find at `\Sylius\Bundle\PromotionBundle\Validator\Constraints\PromotionSubjectCoupon`, but crafted for our command instead of order class(or `PromotionCouponAwarePromotionSubjectInterface` to be exact).